File: Image/automation.py
import pyautogui
import sys
import time
import json

def click_and_type(x, y, text, options=None):
    if options is None:
        options = {}
    
    delay = options.get('delay', 0.1)
    clear_first = options.get('clearFirst', False)
    double_click = options.get('doubleClick', False)
    
    try:
        # Disable pyautogui failsafe for automation
        pyautogui.FAILSAFE = False
        
        # Move and click
        pyautogui.moveTo(x, y)
        
        if double_click:
            pyautogui.doubleClick()
        else:
            pyautogui.click()
        
        # Wait for focus
        time.sleep(delay)
        
        # Clear existing text if requested
        if clear_first:
            pyautogui.hotkey('ctrl', 'a')
            time.sleep(0.05)
        
        # Type text
        pyautogui.typewrite(text)
        
        return {"status": "success", "message": "Automation completed"}
        
    except Exception as e:
        return {"status": "error", "message": str(e)}

if __name__ == "__main__":
    if len(sys.argv) < 4:
        result = {"status": "error", "message": "Usage: python automation.py <x> <y> <text> [options_json]"}
    else:
        x = int(sys.argv[1])
        y = int(sys.argv[2])
        text = sys.argv[3]
        options = json.loads(sys.argv[4]) if len(sys.argv) > 4 else {}
        
        result = click_and_type(x, y, text, options)
    
    print(json.dumps(result))

File: Image/automation.js
const { spawn } = require('child_process');
const path = require('path');

class DesktopAutomator {
    constructor() {
        this.pythonScript = path.join(__dirname, 'automation.py');
    }
    
    async clickAndType(x, y, text, options = {}) {
        return new Promise((resolve, reject) => {
            const optionsJson = JSON.stringify(options);
            const args = [this.pythonScript, x.toString(), y.toString(), text, optionsJson];
            
            const python = spawn('python', args);
            let output = '';
            let error = '';
            
            python.stdout.on('data', (data) => {
                output += data.toString();
            });
            
            python.stderr.on('data', (data) => {
                error += data.toString();
            });
            
            python.on('close', (code) => {
                if (code !== 0) {
                    reject(new Error(`Python script failed: ${error}`));
                    return;
                }
                
                try {
                    const result = JSON.parse(output.trim());
                    if (result.status === 'success') {
                        resolve(result);
                    } else {
                        reject(new Error(result.message));
                    }
                } catch (parseError) {
                    reject(new Error(`Failed to parse Python output: ${output}`));
                }
            });
        });
    }
}

// Usage
async function main() {
    const automator = new DesktopAutomator();
    
    try {
        // Simple usage
        await automator.clickAndType(500, 300, "Hello World!");
        console.log('First automation completed');
        
        // With options
        await automator.clickAndType(400, 200, "New text", {
            delay: 0.2,
            clearFirst: true,
            doubleClick: false
        });
        console.log('Second automation completed');
        
    } catch (error) {
        console.error('Automation failed:', error);
    }
}

main();